### PR TITLE
add api docs

### PR DESCRIPTION
**Pull request by tyler-whitman**
Add api docs using https://typedoc.org/
auto deploys to docs branch using https://github.com/JamesIves/github-pages-deploy-action and github actions

You can regenerate them by running `yarn docs`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -197,5 +197,5 @@ client.transactions.submit(signedPaymentTxn.toString())
 
 ```
 
-> :warning: Note that oracle prices change over time. It's possible for a transaction to fail if the oracle price changes in between the time the transaction is conrstructed and when it is absorbed by the consensus group. The API exposes what the next oracle price will be at `https://api.helium.io
+> ⚠️ Note that oracle prices change over time. It's possible for a transaction to fail if the oracle price changes in between the time the transaction is conrstructed and when it is absorbed by the consensus group. The API exposes what the next oracle price will be at `https://api.helium.io
```
Comment (matthewcarlreetz) [inline README.md:200]: Little typo here "conrstructed" should be "constructed".

Comment (tyler-whitman) [reply README.md]: fixed

**Comment by allenan** [comment]
is there any way to have this build on demand, rather than committing to source?

**Comment by allenan** [comment]
maybe some typedoc/pages github action integration?

**Comment by tyler-whitman** [comment]
> is there any way to have this build on demand, rather than committing to source?

Yeah let me look into it. I was using it to host with GH pages easily but maybe there is a better way.

**Review by allenan — APPROVED** [review]

**Pull request merged**